File: frontend/src/types/business.ts
/**
 * Business-related type definitions
 */
import type { BaseEntity } from './common';

export interface Department extends BaseEntity {
  number: string;
  name: string;
}

export interface DepartmentCreate {
  number: string;
  name: string;
}

export interface Location extends BaseEntity {
  number: string;
  name: string;
}

export interface LocationCreate {
  number: string;
  name: string;
}

export interface Manufacturer extends BaseEntity {
  name: string;
  logo_file?: string;
  sorting: number;
}

export interface ManufacturerCreate {
  name: string;
  logo_file?: string;
  sorting: number;
}

export interface Team extends BaseEntity {
  name: string;
  description?: string;
}

export interface TeamCreate {
  name: string;
  description?: string;
}

export interface Warehouse extends BaseEntity {
  warehouse_name: string;
  number: string;
  markup: number;
}

export interface WarehouseCreate {
  warehouse_name: string;
  number: string;
  markup: number;
}

export interface Commission extends BaseEntity {
  type: string;
  percentage: number;
  gp: boolean;
  sales: boolean;
  commercial_billing: boolean;
  payment: boolean;
}

export interface CommissionCreate {
  type: string;
  percentage: number;
  gp: boolean;
  sales: boolean;
  commercial_billing: boolean;
  payment: boolean;
}

File: backend/models/business.py
from pydantic import BaseModel
from typing import Optional

class Department(BaseModel):
    id: Optional[int] = None
    number: str
    name: str

class DepartmentCreate(BaseModel):
    number: str
    name: str

class Location(BaseModel):
    id: Optional[int] = None
    number: str
    name: str

class LocationCreate(BaseModel):
    number: str
    name: str

class Manufacturer(BaseModel):
    id: Optional[int] = None
    name: str
    logo_file: Optional[str] = None
    sorting: int = 0

class ManufacturerCreate(BaseModel):
    name: str
    logo_file: Optional[str] = None
    sorting: int = 0

class Team(BaseModel):
    id: Optional[int] = None
    name: str
    description: Optional[str] = None

class TeamCreate(BaseModel):
    name: str
    description: Optional[str] = None

class Warehouse(BaseModel):
    id: Optional[int] = None
    warehouse_name: str
    number: str
    markup: float = 0.00

class WarehouseCreate(BaseModel):
    warehouse_name: str
    number: str
    markup: float = 0.00

class Commission(BaseModel):
    id: Optional[int] = None
    type: str
    percentage: float = 0.00
    gp: bool = False
    sales: bool = False
    commercial_billing: bool = False
    payment: bool = False

class CommissionCreate(BaseModel):
    type: str
    percentage: float = 0.00
    gp: bool = False
    sales: bool = False
    commercial_billing: bool = False
    payment: bool = False
